app/js/FileConverter.js

```diff
@@ -1,133 +1,116 @@
+    '-density',
+    '300',
+    '-define',
+    `pdf:fit-page=${width}`,
+    `${sourcePath}[0]`,
+    '-resize',
+    width
+  ])
+}
 
-  thumbnail(sourcePath, callback) {
-    const destPath = `${sourcePath}.png`
-    sourcePath = `${sourcePath}[0]`
-    const width = '260x'
-    let command = [
-      'convert',
-      '-flatten',
-      '-background',
-      'white',
-      '-density',
-      '300',
-      '-define',
-      `pdf:fit-page=${width}`,
-      sourcePath,
-      '-resize',
-      width,
-      destPath
-    ]
-    logger.log({ sourcePath, destPath, command }, 'thumbnail convert file')
-    command = Settings.commands.convertCommandPrefix.concat(command)
-    return safe_exec(command, childProcessOpts, function(err, stdout, stderr) {
-      if (err != null) {
-        logger.err(
-          { err, stderr, sourcePath },
-          'something went wrong converting file to thumbnail'
-        )
-      } else {
-        logger.log({ sourcePath, destPath }, 'finished thumbnailing file')
-      }
-      return callback(err, destPath)
+async function preview(sourcePath) {
+  const width = '548x'
+  return convert(sourcePath, 'png', [
+    'convert',
+    '-flatten',
+    '-background',
+    'white',
+    '-density',
+    '300',
+    '-define',
+    `pdf:fit-page=${width}`,
+    `${sourcePath}[0]`,
+    '-resize',
+    width
+  ])
+}
+
+async function _convert(sourcePath, requestedFormat, command) {
+  logger.log({ sourcePath, requestedFormat }, 'converting file format')
+
+  if (!APPROVED_FORMATS.includes(requestedFormat)) {
+    throw new ConversionError({
+      message: 'invalid format requested',
+      info: { format: requestedFormat }
     })
-  },
+  }
 
-  preview(sourcePath, callback) {
-    logger.log({ sourcePath }, 'preview convert file')
-    const destPath = `${sourcePath}.png`
-    sourcePath = `${sourcePath}[0]`
-    const width = '548x'
-    let command = [
-      'convert',
-      '-flatten',
-      '-background',
-      'white',
-      '-density',
-      '300',
-      '-define',
-      `pdf:fit-page=${width}`,
-      sourcePath,
-      '-resize',
-      width,
-      destPath
-    ]
-    command = Settings.commands.convertCommandPrefix.concat(command)
-    return safe_exec(command, childProcessOpts, function(err, stdout, stderr) {
-      if (err != null) {
-        logger.err(
-          { err, stderr, sourcePath, destPath },
-          'something went wrong converting file to preview'
-        )
-      } else {
-        logger.log(
-          { sourcePath, destPath },
-          'finished converting file to preview'
-        )
-      }
-      return callback(err, destPath)
+  const timer = new metrics.Timer('imageConvert')
+  const destPath = `${sourcePath}.${requestedFormat}`
+
+  command.push(destPath)
+  command = Settings.commands.convertCommandPrefix.concat(command)
+
+  try {
+    await safeExec(command, {
+      killSignal: KILL_SIGNAL,
+      timeout: FOURTY_SECONDS
     })
+  } catch (err) {
+    logger.err(
```

Comment:
Do we need this log here, if we pass the same info to the error constructor below? Will it be logged fully at the top level anyway? :)

Reply:
No, good call. I think if we're using `OError` we can remove these error logs.